android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java

```diff
@@ -705,18 +705,20 @@ public String getSecurityPatchSync() {
   @ReactMethod
   public void getSecurityPatch(Promise p) { p.resolve(getSecurityPatchSync()); }
 
-  @ReactMethod
-  public void getUserAgent(Promise p) {
+  @ReactMethod(isBlockingSynchronousMethod = true)
+  public String getUserAgentSync() {
```

Comment:
This will be a very slow synchronous method :-), but I see no harm in having it of course - users have their own needs